### PR TITLE
Add blog post about September SIP Meeting results

### PR DESCRIPTION
**Pull request by jvican**

### CODE REVIEW

**Comment by heathermiller** [comment]
Hey @dragos and @jsuereth could you guys provide your written feedback so we can publish this? We're waiting on all reviewers to release their feedback before we publish.

**Comment by jvican** [comment]
Hey @soc, thanks for asking. I'm the SIP Process lead.
  
70% of the Committee is 4.2 people. Here is what [the regulations](http://docs.scala-lang.org/sips/sip-submission.html) say:
  
> For a SIP to be accepted, it must fulfill two requirements:
>  
>    - 70% of the quorum votes in favor of it.
>    - Martin Odersky does not veto it.

The regulations don't specify how rounding works. I took the liberty to round to the closest integer, which is 4 members in this case.
  
I admit that the regulations should be more clear on this detail. I will make sure to follow up and clarify the language.

File: blog/_posts/2016-09-20-sip-meeting-september-results.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    ago, when the proposal was numbered, Sébastien Doeraene convinced the
+    Committee to try to improve the performance hit in the provided
+    implementation, for which he asked for two months. Unfortunately, after some
+    work during two months, a 6% performance hit remained. As a result,
+    Sébastien proposed to reject the proposal, whose motion was unanimously
+    approved by the Committee. For more information, have a look at his full
+    report [here](https://github.com/scala/scala.github.com/pull/548).
+-   [SIP-27: Trailing Commas](http://docs.scala-lang.org/sips/pending/trailing-commas.html).
+		The trailing commas proposal was acknowledged and numbered by the Committee
+		three months ago. This time, Dale Wijnand, his author, was invited as a
```
Comment (dwijnand) [inline blog/_posts/2016-09-20-sip-meeting-september-results.md]: s/his author/the author/

Comment (jvican) [inline blog/_posts/2016-09-20-sip-meeting-september-results.md]: :+1:

File: blog/_posts/2016-09-20-sip-meeting-september-results.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+Scala developer with interesting ideas to
+[participate](http://docs.scala-lang.org/sips/sip-submission.html)!
+
+Next month, we’ll discuss four more proposals, including
+[SIP-20: Improved lazy val initialization](http://docs.scala-lang.org/sips/pending/improved-lazy-val-initialization.html) and
+[SIP-24: Repeated by-name parameters](http://docs.scala-lang.org/sips/pending/repeated-byname.html).
+If you want to read all our previous minutes, head out to [the official website](http://docs.scala-lang.org/sips/minutes-list.html).
+
+Did you know that each month, we conduct these SIP meetings
+on-air? You can tune in and ask questions to the SIP committee, and have
```
Comment (SethTisue) [inline blog/_posts/2016-09-20-sip-meeting-september-results.md]: about live questions, I don't think we should allow/encourage this, actually. Community feedback should be provided in advance, it's too distracting to have it come in in the middle of a meeting. And in practice, we aren't actually doing it anyway.

File: blog/_posts/2016-09-20-sip-meeting-september-results.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    Originally proposed in 2013, spores are functions that allow programmers to
+    control the captured environment. Among several advantages, they allow
+    coarse-grained control over safe closure serialization. This proposal,
+    reviewed by Martin Odersky, was in its first review iteration. A follow-up
+    iteration will take place within two months, when the authors (Heather
+    Miller and Philipp Haller) are expected to update the proposal and look into
+    a more thorough environment analysis.  Martin and Seth have left their
+    impressions on spores [here](http://disq.us/p/1c66wxe).
+-   [SIP-26: Unsigned Integer Data Types](http://docs.scala-lang.org/sips/rejected/unsigned-integers.html).
+    Proposed by Sébastien Doeraene and Denys Shabalyn, unsigned integers have
```
Comment (adriaanm) [inline blog/_posts/2016-09-20-sip-meeting-september-results.md]: typo: "Denys Shabalin"

**Comment by dwijnand** [comment]
@soc 

I'm reasonably sure that if @jsuereth hadn't left early he would've also voted accept, pushing the vote over 70%.

But then if @adriaanm had been present I would guess he would have voted against (just a guess). I'm not sure about @som-snytt having never seen him weigh in on the issue (to my recollection).

It does bring an interesting dynamic about who is present the day of the decision. So given how few yet very significant each vote is I wonder if it's perhaps a better representation of the committee's decision if **everyone** on the committee votes, even if it means the decision is revealed a few days later. The meetings are few and far between (once a month) so it's viable, and given how important the decisions are for the future of the language perhaps it's an improvement over the current process.

**Comment by sjrd** [comment]
> So given how few yet very significant each vote is I wonder if it's perhaps a better representation of the committee's decision if everyone on the committee votes, even if it means the decision is revealed a few days later.

:+1:

**Comment by som-snytt** [comment]
Because I was at-summoned, I must contribute: I left the SIP process b/c agoraphobia or what have you, though to my credit I did fix my laptop mic problem under Windows 10.

I've published an opinion on trailing commas elsewhere: If people say, This makes life easier, then people who aren't actively harmed by the proposal should say, OK, I can live with that.

To me, that's what community means.

So I was +1 on the SIP. I would have preferred trailing commas everywhere there are commas.

As a footnote, to me, the Scala ethos is that you can do OOP with FP, an impossible compromise, because we are generous, because we try to enable new paradigms, things you couldn't do otherwise. Odersky said that at some point: it's not for the sake of conservative safety, but to enable technology that is otherwise too dangerous or risky.

I think Heidegger said something similar on what "conservative" means. It means creating the possibility of something daring.

My other thought on trailing commas is that there could be a `spp` scala pre-processor to do trivial pre-processing which doesn't amount to a fork of scala but accounts for trivial syntax preferences. Similarly, I've done scripting as a post-processor to parsing. If it's just syntax, then pre- or post- process it.

**Comment by refried** [comment]
I don’t know anything about the SIP, just chiming in to suggest changing the number to 65% if that’s high enough.  Or “at least one more than the number required for a simple majority”.

**Comment by dwijnand** [comment]
With the current rules for a proposal to be accepted it requires at least either 6/8 (75%), 5/7 (71%) or 5/6 (83%) accept votes.

With your proposal that changes the last one to 4/6 (67%), which is a simple majority.

But I do like the idea of changing the rule to "at least one more than the number required" instead of trying to pick a good percentage between 51-100%.

**Comment by soc** [comment]
@refried I think the numbers should be much closer to 100% than to 70%. This is not some kind of election that you can repeat every few years and start from scratch again. Adding things has a high cost, and deprecating/removing even more so.

**Comment by refried** [comment]
@soc That’s fine -- my suggestion was just to choose a policy and then abide by it.  Not a very helpful suggestion, really.

**Comment by dwijnand** [comment]
@soc No progress will be made if you require 100%.

Think of the right-biased Either change, and how much back and forth happened there. If that had required a 100% acceptance it would have never happened.

**Comment by soc** [comment]
For me, the quality of proposals is more important than the quantity. I'd rather have three good proposals rejected than a bad one accepted.

(Also, I didn't say "require 100%", I said "closer to 100% than to 70%".)

I believe that rough consensus and running code will always be superior to committees, voting and paperwork.

**Comment by som-snytt** [comment]
Maybe it really is an election. There should be term limits for terms!

Instead of looking at deprecations, features should need to be re-elected every four years. Every two terms, they must be forcibly upgraded to the next new thing, or booted out of office.

You don't actually have to hold an election. Just monitor download stats and feature usage as reported by instrumented compilers in the field.

I also agree that making voting a judgment call is the first step toward disenfranchisement. I almost wrote disenchantment.

**Comment by dwijnand** [comment]
When you have 8 votes the cut-offs above 70% are 75%, 87.5% and 100%, so I just assumed you were after 100%. Were you thinking 87.5%? Ie. at most one person can disagree?

**Comment by soc** [comment]
Something like high 80% to low 90% seems to be about right. How that turns into persons-that-can-disagree will always vary, e. g. at the moment we are down two members, as all the community representatives have left the committee.

**Comment by SethTisue** [comment]
This PR has been sitting for a while now. I suggest revising it to say that the result of the vote was unclear and it'll get settled at the next meeting, and then go ahead and publish.

**Comment by SethTisue** [comment]
@jvican we need to do _something_ here. what do you suggest?

**Comment by jvican** [comment]
@SethTisue I suggest we leave it open just in case some other community member wants to comment on the voting system that we will discuss in one week. I'm sure we can still great valuable insights. Regarding the release of this post, I would rather announce everything after our next SIP meeting (next Wednesday).

File: blog/_posts/2016-09-20-sip-meeting-september-results.md (hunk 2 above)
Comment (jvican) [inline blog/_posts/2016-09-20-sip-meeting-september-results.md]: I agree Community feedback should be provided in advance. I think I have a formula for this... I'll try it out soon.

**Comment by SethTisue** [comment]
ok, thanks for the update

**Comment by dwijnand** [comment]
For anyone here and not in the Gitter rooms, the next SIP meeting is tomorrow, Tuesday, not Wednesday.

**Comment by soronpo** [comment]
Sorry I missed commenting here prior to the meeting, but I think I have a simplified formula for SIP acceptance. I also posted this here: https://groups.google.com/forum/#!topic/scala-sips/Pn2-hpFrwfA

```
P = Number of votes in favor
N = Number of votes opposed
T = Minimum for acceptance
(People can abstain)
```

So to accept the proposal the following formula must be true:
```
P - N > T
```

`T` is set as the required filter by the committee.
Suggestion: select `T` as half of the committee members.
Note: `T` can also be a constant

The logic behind this is that enough people should care to vote in favor, while not enough people care to vote against.


Example:
Committee of 8, so `T = 4`

- Proposal A: 4 persons in favor, none against, the rest have no opinion and abstain. The proposal is rejected.
- Proposal B: 5 persons in favor, none against, the rest have no opinion and abstain. The proposal is accepted.
- Proposal C: 5 persons in favor, 1 against, the rest have no opinion and abstain. The proposal is rejected.


Rounding when calculating `T`:
Considering that the condition is `>` in the formula, it is suggested that if indeed `T` is selected as half the members, then the number will be rounded down.
E.g: Number of committee members is 9, then T will be 4. So if 5 persons in favor with none against will accept the proposal.

**Comment by jvican** [comment]
Thanks for the time invested in your proposal @soronpo!
  
I propose the discussion continues here: https://dev.scala-lang.org/t/sip-simplified-voting-acceptance-formula-suggestion/136.

**Comment by SethTisue** [comment]
@jvican closing just to get it out of the queue. assuming this in your own todo list somewhere

**Pull request closed without merging**